### PR TITLE
API: library is open

### PR DESCRIPTION
**Pull request by rerowep**
* NEW: library is_open function

Signed-off-by: Peter Weber <Peter.Weber@rero.ch>

open the console and try to test the new date functions from library

### CODE REVIEW

File: rero_ils/modules/libraries/api.py [hunk 1]
```diff
@@ -58,3 +70,159 @@ def get_all_library_names(cls):
...
+            times_open = times_open or (time_to_test >= start_time) and \
+                                       (time_to_test <= end_time)
+        return times_open
+
+    def _is_in_period(self, datetime_to_test, exception_date):
+        """Test if date is period."""
+        start_date = parser.parse(exception_date['start_date'])
+        end_date = exception_date.get('end_date')
+        if end_date:
+            end_date = parser.parse(exception_date['end_date'])
```
Comment (jma) [inline rero_ils/modules/libraries/api.py]: `parser.parse(end_date)`, no?

File: rero_ils/modules/libraries/api.py [hunk 2]
```diff
@@ -58,3 +70,159 @@ def get_all_library_names(cls):
...
+                open=times_open,
+                date=date,
+                exception_dates=exception_dates,
+                day_only=day_only
+            )
+            if exception != times_open:
+                times_open = not times_open
+        return times_open
+
+    def next_open(self, date=datetime.now(), previous=False):
```
Comment (jma) [inline rero_ils/modules/libraries/api.py]: not sure that default parameter calling function works... i.e. `datetime.now()`.

File: rero_ils/modules/libraries/api.py (hunk 1 above)
Comment (rerowep) [reply rero_ils/modules/libraries/api.py]: Yes, is better like this

File: rero_ils/modules/libraries/api.py (hunk 2 above)
Comment (rerowep) [reply rero_ils/modules/libraries/api.py]: I have tested it. It is working.

**Pull request merged**